kv/kvserver/loqrecovery: TestRetrieveRangeStatus failed

kv/kvserver/loqrecovery.TestRetrieveRangeStatus [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11515632?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11515632?buildTab=artifacts#/) on master @ [91907b3e326ae42effdfcac43e0bad0ea2d33d3c](https://github.com/cockroachdb/cockroach/commits/91907b3e326ae42effdfcac43e0bad0ea2d33d3c):


```
=== RUN   TestRetrieveRangeStatus
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/ea0927e04eade2740899effbae326bb2/logTestRetrieveRangeStatus3355417607
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:121: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
        (override via TestingSetDefaultTenantSelectionOverride)
    server_integration_test.go:520: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/kv/kvserver/loqrecovery_test/pkg/kv/kvserver/loqrecovery/server_integration_test.go:520
        	Error:      	Should be empty, but was operation "retrieve ranges health" timed out after 1m0.001s (given timeout 1m0s): aborted during DistSender.Send: context deadline exceeded
        	Test:       	TestRetrieveRangeStatus
        	Messages:   	should have no error
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/ea0927e04eade2740899effbae326bb2/logTestRetrieveRangeStatus3355417607
--- FAIL: TestRetrieveRangeStatus (65.55s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/replication
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRetrieveRangeStatus.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-30997

Another failure mode that I encountered while running this test locally:

```
--- FAIL: TestRetrieveRangeStatus (63.65s)
    test_log_scope.go:167: test logs captured to: /Users/pavel/go/src/github.com/cockroachdb/cockroach/tmp/_tmp/c87d7e65c1e00835d84bcde1797d3f9a/logTestRetrieveRangeStatus2013138544
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:121: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
        (override via TestingSetDefaultTenantSelectionOverride)
    server_integration_test.go:518: failed to find scratch range in unavailable ranges
    panic.go:522: -- test log scope end --
FAIL

...
//pkg/kv/kvserver/loqrecovery:loqrecovery_test                           FAILED in 64.0s
```

--- a/pkg/kv/kvserver/loqrecovery/server_integration_test.go
+++ b/pkg/kv/kvserver/loqrecovery/server_integration_test.go
@@ -485,7 +485,8 @@ func TestRetrieveRangeStatus(t *testing.T) {
 
 	// Use scratch range to ensure we have a range that loses quorum.
 	sk := tc.ScratchRange(t)
-	require.NoError(t, tc.WaitForFullReplication(), "failed to wait for full replication")
+	require.NoError(t, tc.WaitFor5NodeReplication(),
+		"failed to wait for full replication of 5 node cluster")
 	tc.ToggleReplicateQueues(false)
 	tc.SplitRangeOrFatal(t, testutils.MakeKey(sk, []byte{255}))
 

